tests/10apidoc/12device_management.pl

```diff
@@ -209,11 +211,16 @@ sub matrix_set_device_display_name {
             ->main::expect_http_404;
       })->then( sub {
          # our access token should be invalidated
-         do_request_json_for(
-            $other_login,
-            method  => "GET",
-            uri     => "/r0/sync",
-         )->main::expect_http_401;
+         try_repeat_until_success {
+            do_request_json_for(
+               $other_login,
+               method  => "GET",
+               uri     => "/r0/sync",
+            )->main::expect_http_401
+            ->else_with_f( sub {
+               my ( $f ) = @_; delay( 1 )->then( sub { $f } );
```

Comment:
Do you want to wait an entire second, at least the first time? If you happen to just lose a race by mere milliseconds, it might be nice to start with smaller waits initially. Or use a fixed smaller amount?